internal/databases/mysql/mysql.go

```diff
@@ -271,6 +271,9 @@ func getLogsCoveringInterval(folder storage.Folder, start time.Time, includeStar
 	})
 	var logsToFetch []storage.Object
 	for _, logFile := range logFiles {
+		if logFile.GetLastModified().After(endBinlogTS) {
+			continue // don't fetch binlogs from future
+		}
```

Comment:
This shold not affect the final result, as mysqbinlog started as
```
mysqlbinlog --stop-datetime="$WALG_MYSQL_BINLOG_END_TS"
```
and should not replay transactions happened after `fetchUntilTS`

Reply:
This patch fixes following issue:

Live MySQL cluster **A** constantly uploads binlogs. We are trying to restore new cluster (let's call it **B**) from S3 on some point in time, e.g. `NOW()` (or future time).
Then we can have following scenario:
'Fast' servers restore MySQL on Snapshot + binlog.123 (Binlog 123 is the latest observed binlog for 'fast' hosts).
'Slow' servers may restore MySQL on Snapshot+binlog.124 (Becase cluster **A** have uploaded one more binlog during 'slow' restore).
As a result we have different GTIDs in a cluster. It is easier to prevent such scenario rather than fixing it afterwads.

To prevent different GTID-sets we should:
 * introduce hard limits on which binlogs we are replaying
 * point in time for this limits should be `NOW()` or in past.